src/worker/tasks/chainIndexer.ts

```diff
@@ -1,135 +1,122 @@
-import { StaticJsonRpcBatchProvider } from "@thirdweb-dev/sdk";
-import { Address } from "thirdweb";
+import {
+  eth_blockNumber,
+  eth_getBlockByNumber,
+  getRpcClient,
+  type Address,
+} from "thirdweb";
 import { getBlockForIndexing } from "../../db/chainIndexers/getChainIndexer";
 import { upsertChainIndexer } from "../../db/chainIndexers/upsertChainIndexer";
 import { prisma } from "../../db/client";
 import { getContractSubscriptionsByChainId } from "../../db/contractSubscriptions/getContractSubscriptions";
-import { getSdk } from "../../utils/cache/getSdk";
+import { getChain } from "../../utils/chain";
 import { logger } from "../../utils/logger";
+import { thirdwebClient } from "../../utils/sdk";
 import { ProcessEventsLogQueue } from "../queues/processEventLogsQueue";
 import { ProcessTransactionReceiptsQueue } from "../queues/processTransactionReceiptsQueue";
 
-export const createChainIndexerTask = async (args: {
-  chainId: number;
-  maxBlocksToIndex: number;
-  toBlockOffset: number;
-}) => {
-  const { chainId, maxBlocksToIndex, toBlockOffset } = args;
+// A reasonable block range that is within RPC limits.
+// The minimum job time is 1 second, so this value should higher than the # blocks per second
+// on any chain to allow catching up if delayed.
+const MAX_BLOCK_RANGE = 500;
 
-  const chainIndexerTask = async () => {
```

Comment:
Changes:
- This function no longer returns a function, it just executes the logic.
- Remove try/catch, already caught above.
- Move to v5 SDK